jwtauth.go

```diff
@@ -120,6 +137,10 @@ func VerifyToken(ja *JWTAuth, tokenString string) (jwt.Token, error) {
 }
 
 func (ja *JWTAuth) Encode(claims map[string]interface{}) (t jwt.Token, tokenString string, err error) {
+	if ja.keySet != nil {
+		return nil, "", fmt.Errorf("encode not supported")
```

Comment:
I'm working on rebasing and adding more documentation.  I don't understand why encode isn't supported in this case and I think we need a better error message for it.  Can you explain what this is for?

Reply:
I think that's the wrong error message probably from some copy-pasta. I think the message should just reflect that the JWKS isn't set so nothing can be encoded here.

Reply:
Based on my understanding, I'm considering updating Encode like this:

```go

Reply:
Looks fine to me. I was reading the code wrong and was thinking the `ja.keySet` was an error. I'll have to go back and look at the entire PR to recall the context why this was done.

Reply:
I'm refactoring it a bit now that I understand it better.